Refactor QueryExtensions to move the implementation to API

alecslupu opened the pull request:
#### :tophat: What? Why?
In this PR I am refactoring the `Decidim::Api::QueryType` to include by default the methods defined in `decidim-core`. The rationale behind this change is that `decidim-api`, but `decidim-core` could work without api.

Since we are moving methods in this module, i am disabling bullet to avoid errors.

#### Testing
1. Pipeline should be green

:hearts: Thank you!

alecslupu committed:
Refactor QueryType to API

alecslupu committed:
Remove bullet error

alecslupu committed:
Remove unneded autoload

alecslupu committed:
Fix typo

alecslupu committed:
Disable bullet

andreslucena commented:
> Since we are moving methods in this module, i am disabling bullet to avoid errors.

Which bullet errors? Can you fix them? Or they're too many? 

I'm trying to check them out in the CI logs but I can't find them. Can you provide them please?

alecslupu commented:
> > Since we are moving methods in this module, i am disabling bullet to avoid errors.
> 
> Which bullet errors? Can you fix them? Or they're too many?
> 
> I'm trying to check them out in the CI logs but I can't find them. Can you provide them please?

I have built this PR against a personal fork where i am building the API stuff, and i got the error below.  

<img width="1411" height="373" alt="image" src="https://github.com/user-attachments/assets/1a6e4f0a-84ef-4f0e-897b-c2b8e15711b8" />

andreslucena commented:
> I have built this PR against a personal fork where i am building the API stuff, and i got the error below.

Is that the full stacktrace?? Can you provide an actual URL??? 

If it is a real error, we need to remove that eager loading. If it is a false positive, we need to configure Bullet to let it know by using the `add_safelist` method: https://github.com/flyerhzm/bullet#safe-list 

We need to actually fix the broken windows and not just look to another side. If us as maintainers do not pay the technical debt, we will have a big slow software that will be impossible to maintain.

alecslupu committed:
Fix bullet error

alecslupu commented:
> Is that the full stacktrace??

https://github.com/tremend-cofe/decidim-api/actions/runs/20004940318/job/57365738778


> Is that the full stacktrace?? Can you provide an actual URL???

The stacktrace from local is as follows: 

<img width="1310" height="282" alt="image" src="https://github.com/user-attachments/assets/0393d011-c93e-4cd1-b176-1a8ca0472126" />

The `includes` part has been added via #8417, before bullet was a thing in decidim ( #11416 ). 
Also, on decidim_core workflow, the `bullet_unused_eager_loading` is set to false, and this is why it has not been revealed before. I have ran locally the conversations specs, and there was no error with bullet fully enabled. 

https://github.com/decidim/decidim/blob/develop/.github/workflows/ci_core.yml#L50C1-L50C41

To fully enable bullet, we'd need patch a lot of specs so that we add additional records, or recreate as much as possible the real environment. When we will have all the pipelines enabled, we'd have some error contradicting ... in some places we'll have "UnoptimizedQuery " / "Avoid eager load", and in some places we'll have "Avoid N+1 by adding include" ... so it will be fun to fully enable bullet. And if we add the exceptions to "safe list", it would be just obsolete, like having it disabled ... 

 In the meantime, i have reverted the github config, and i have removed the includes.

andreslucena commented:
>  Also, on decidim_core workflow, the bullet_unused_eager_loading is set to false, and this is why it has not been revealed before. I have ran locally the conversations specs, and there was no error with bullet fully enabled.

Awesome! No need to have the eager loading then, right? So Bullet was right? 

> To fully enable bullet, we'd need patch a lot of specs so that we add additional records, or recreate as much as possible the real environment. 

I understand that it is out of the scope of this current PR for sure, my critique was that at least let's try to not make the checks even worse than what we already have

alecslupu commented:
> Awesome! No need to have the eager loading then, right? So Bullet was right?

I still do not know this ... It might have been right, or it may be something related to the storage provider, or it just have been fixed somewhere between #8417 and #15768. 
The Bullet config is just something that i need to throw in my backlog ...
